### DISCUSSION TITLE
Self-hosted integrations

### DISCUSSION THREAD

**Discussion by kalexmills** [cuelang/cue#792] [Q&A]
I am noticing that several of the current set of integrations (currently) requires external tooling and/or native tool support. It will be interesting to see how this approach will scale as the industry evolves.

I wonder if any thought has been given towards the feasibility of writing self-hosted integrations in Cue itself -- or if the Cue language is even planned to become powerful enough for that to be a reasonable thing to expect.

Thanks in advance.

**Comment by mpvl** [comment]
The idea is at some point to have a better bridge to other languages like Go and Python to allow for a more pluggable architecture.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/792.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
